### ISSUE TITLE
"Search" and "likes" mode do not use all available APIs

### ISSUE THREAD

**Issue by nadesai** [eleurent/twitter-graph#25]
Both modes only pass `apis[0]` into the relevant method, even when multiple APIs are available.

For "search" mode, this may make sense, since the number of tweets returned by the search is likely bounded by the 7-day search limit. However, for a particularly popular query and a high `max_count`, this bound may be very high and API reuse could help speed up the query.

There is no time bound on API queries for favorites, and here it definitely makes sense to allow for multiple API usage.

**Comment by eleurent** [comment]
I completely agree! It should be quite straightforward to implement, I think, similarly to what is done for friendships.
